src/Stack/GhcPkg.hs

```diff
@@ -167,6 +168,21 @@ unregisterGhcPkgId wc cv pkgDb gid ident = do
                 [packageIdentifierString ident]
             _ -> ["--ipid", ghcPkgIdString gid])
 
+-- | unregister list of package ghcids, available from GHC 8.0.1
+unregisterGhcPkgIds :: (HasProcessContext env, HasLogFunc env)
+                    => WhichCompiler
+                    -> Path Abs Dir -- ^ package database
+                    -> [GhcPkgId]
+                    -> RIO env ()
+unregisterGhcPkgIds wc pkgDb gids = do
+    eres <- ghcPkg wc [pkgDb] args
+    case eres of
+        Left e -> logWarn $ displayShow e
+        Right _ -> return ()
+  where
+    args = "unregister" : "--user" : "--force" :
+        concatMap (\gid -> ["--ipid", ghcPkgIdString gid]) gids
```

Comment:
If you expand the aforementioned `case`, this (duplicate) function then becomes unnecessary, as you can just rename `unregisterGhcPkgId`.